zerver/views/user_groups.py

```diff
@@ -378,15 +388,16 @@ def remove_subgroups_from_group_backend(
 
 
 @require_user_group_edit_permission
-@has_request_variables
+@typed_endpoint
 def update_subgroups_of_user_group(
     request: HttpRequest,
     user_profile: UserProfile,
-    user_group_id: int = REQ(json_validator=check_int, path_only=True),
-    delete: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
-    add: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
+    *,
+    user_group_id: PathOnly[Json[int]],
+    delete: Json[list[int]] | None = None,
+    add: Json[list[int]] | None = None,
 ) -> HttpResponse:
-    if not add and not delete:
+    if add is None and delete is None:
```

Comment:
Why have you made this change? This will no longer be true if `add` and `delete` are empty arrays.

Reply:
But `add` and `delete` are no longer empty arrays as default value, i can't find a place where they are passed as empty arrays

Reply:
Yes, but the user can still pass an empty array.

Reply:
hmm, in case `Pydantic` accepts `list[int]` to also be also just `[]` (right or not ?) then yes you are right .

Reply:
Yes list[int] will allow the user to pass an empty list.